src/components/ViewDonation/ViewDonation.jsx

```diff
@@ -0,0 +1,171 @@
+import { useBackend } from '../../contexts/BackendContext';
+import { useParams, useNavigate } from 'react-router-dom';
+import { useState, useEffect } from 'react';
+import {
+  Card,
+  Heading,
+  CardBody,
+  Stack,
+  Box,
+  Text,
+  StackDivider,
+  Flex,
+  Breadcrumb,
+  BreadcrumbItem,
+  BreadcrumbLink,
+  Button,
+  useToast,
+} from '@chakra-ui/react';
+import { ArrowDownIcon } from '@chakra-ui/icons';
+import NotificationsDrawer from '../NotificationsDrawer/NotificationsDrawer';
+import DownloadCSV from '../../utils/downloadCSV';
+import PropTypes from 'prop-types';
+import classes from './ViewDonation.module.css';
+
+const ViewDonation = () => {
+  const { backend } = useBackend();
+  const { id } = useParams();
+  const toast = useToast();
+  const [donationData, setDonationData] = useState([]);
+  const [businessName, setBusinessName] = useState('');
+  const [notification, setNotification] = useState([]);
+  const navigate = useNavigate();
+
+  const handleDownloadCSV = () => {
+    try {
+      DownloadCSV([id], true);
+      toast({
+        title: 'Downloaded CSV',
+        description: `for ${businessName}`,
+        status: 'success',
+        duration: 3000,
+        isClosable: true,
+      });
+    } catch (error) {
+      toast({
+        title: 'Error Downloading CSV',
+        description: error.message,
+        status: 'error',
+        duration: 3000,
+        isClosable: true,
+      });
+    }
+  };
+
+  const handleViewBusiness = () => {
+    navigate(`/ViewBusiness/${donationData.business_id}`);
+  };
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        const donationResponse = await backend.get(`/donation/${id}`);
+        const businessResponse = await backend.get(`/business/${donationResponse.data[0].business_id}`);
+        setBusinessName(businessResponse.data[0].name);
+        setDonationData(donationResponse.data[0]);
+      } catch (error) {
+        console.error('Error fetching donation data:', error);
+      }
+    };
+
+    const getNotifications = async () => {
+      try {
+        const notificationResponse = await backend.get('/notifications');
+        setNotification(notificationResponse.data);
+      } catch (error) {
+        console.error('Error fetching notifications:', error);
+      }
+    };
+    if (notification.length === 0) {
+      getNotifications();
+    }
+
+    getData();
+  }, []);
```

Comment:
⚠️ **[eslint]** <[react-hooks/exhaustive-deps](https://github.com/facebook/react/issues/14920)> <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br>React Hook useEffect has missing dependencies: 'backend', 'id', and 'notification.length'. Either include them or remove the dependency array.